.github/compilation-check-source.yml

```diff
@@ -60,7 +60,10 @@ on:
 
 jobs:
   build-library:
-    runs-on: macOS-latest
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ macOS-latest, windows-latest, ubuntu-latest ]
```

Comment:
lets use anchor here, just like checkout, upload reports and other reusable parts

Reply:
added